### PR TITLE
bugfix/hubspot-duplicates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
PR to address some duplicate HubSpot records uncovered when testing the model on a new schema. Additionally, added unique and not null tests to the final unified rag output model.

### CODE REVIEW

File: models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql [hunk 1]
```diff
@@ -177,4 +178,5 @@ truncated_comments as (
 )
 
 select *
-from truncated_comments
\ No newline at end of file
+from truncated_comments
+where comment_markdown is not null
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql:182]: I found that there were nearly 300k records in this output model that had no RAG relevant data associated with them. I'm not sure if this is how customer environments will look, but it was resulting in some pretty skewed data when testing on our test schemas internally. I don't see this being an issue as we shouldn't include any deal information that isn't useful for a RAG use case.

File: models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql [hunk 2]
```diff
@@ -45,14 +45,15 @@ engagement_emails as (
         engagement_email.email_to_email,
         engagement_email.email_cc_email,
         engagement_email.email_from_email as commenter_email,
-        contacts.contact_name as commenter_name
+        {{ fivetran_utils.string_agg(field_to_agg="contacts.contact_name", delimiter="','") }} as commenter_name
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql:48]: I found a few cases where there are multiple contacts associated with an engagement email. This resulted in fannout. The stringagg will ensure all parties are included in the resulting data, but also not cause any fannout.

File: models/intermediate/hubspot/int_rag_hubspot__deal_document.sql [hunk 3]
```diff
@@ -84,6 +84,21 @@ engagement_details as (
...
+        deal_name,
+        url_reference,
+        created_on,
+        source_relation,
+        {{ fivetran_utils.string_agg(field_to_agg="distinct engagement_type", delimiter="', '") }} as engagement_type,
+        {{ fivetran_utils.string_agg(field_to_agg="distinct contact_name", delimiter="', '") }} as contact_name,
+        {{ fivetran_utils.string_agg(field_to_agg="distinct created_by", delimiter="', '") }} as created_by,
+        {{ fivetran_utils.string_agg(field_to_agg="distinct company_name", delimiter="', '") }} as company_name
+    from engagement_detail_prep
+    group by 1,2,3,4,5
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/hubspot/int_rag_hubspot__deal_document.sql:99]: Similar to the previous model. The joins in the prev cte are not 1:1. So we need to do some creative aggregating to make sure we retain all the necessary information, but to not cause any fannouts.

File: models/rag__unified_document.sql [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {{
     config(
         materialized='table' if unified_rag.is_databricks_sql_warehouse() else 'incremental',
-        partition_by = {'field': 'most_recent_chunk_update', 'data_type': 'date', 'granularity': 'month'}
-            if target.type not in ['spark', 'databricks'] else ['most_recent_chunk_update'],
+        partition_by = {'field': 'update_date', 'data_type': 'date'}
```
Comment (fivetran-joemarkiewicz) [inline models/rag__unified_document.sql:4]: Ran into some issues with the incremental logic on BQ. These changes helped address those issues although it did require adding a new field (which has been documented and docs regen'd).

File: models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql (hunk 1 above)
Comment (fivetran-avinash) [reply models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql:182]: Sounds good!

File: models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql (hunk 2 above)
Comment (fivetran-avinash) [reply models/intermediate/hubspot/int_rag_hubspot__deal_comment_document.sql:48]: Good catch.

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-joemarkiewicz good catches, lgtm

**Pull request merged**